src/lib.rs

```diff
@@ -308,3 +308,117 @@ pub fn narrowable_abgc(args: TokenStream, input: TokenStream) -> TokenStream {
 
     TokenStream::from(expanded)
 }
+
+#[cfg(feature = "gcmalloc")]
+#[proc_macro_attribute]
+pub fn narrowable_gcmalloc(args: TokenStream, input: TokenStream) -> TokenStream {
+    let args = parse_macro_input!(args as AttributeArgs);
+    let input = parse_macro_input!(input as ItemTrait);
+    if args.len() != 1 {
+        panic!("Need precisely one argument to 'narrowable'");
+    }
+    let struct_id = match &args[0] {
+        NestedMeta::Meta(m) => m.name(),
+        NestedMeta::Literal(_) => panic!("Literals not valid attributes to 'narrowable'")
+    };
+    let trait_id = &input.ident;
+    let expanded = quote! {
+        /// A narrow pointer to #trait_id.
+        pub struct #struct_id {
+            // This struct points to a vtable pointer followed by an object. In other words, on a
+            // 64 bit machine the layout is (in bytes):
+            //   0..7: vtable
+            //   8..: object
+            // This is an inflexible layout, since we can only support structs whose alignment is
+            // the same or less than a usize's.
+            // FIXME: This is somewhat a hangover from abgc. Why can't it
```

Comment:
I no longer remember why. I'm not sure this FIXME is particularly useful though?

Reply:
The FIXME should have been removed once I upgraded the PR from draft to merge-able. Fixed: 93b2b899fcb6c0cb8c3f618676c1848ae2c2afc9